source/module_io/output_mat_sparse.cpp

```diff
@@ -0,0 +1,81 @@
+#include "output_mat_sparse.h"
+
+#include "cal_r_overlap_R.h"
+#include "write_HS_R.h"
+
+namespace ModuleIO
+{
+
+Output_Mat_Sparse::Output_Mat_Sparse(int out_mat_hsR,
```

Comment:
I'm really confused about what this function does, it can output 4 different matrixs, which need different inputs. 
Why do you use this function to gather these 4 functions together?

Reply:
@dyzheng ISSUE #2532 required to do this:
"The same things occur from line 1066 to line 1095 (combine out_mat_hsR, out_mat_t, out_mat_dh, and out_mat_r into one interface function)." as raised by @mohanchen